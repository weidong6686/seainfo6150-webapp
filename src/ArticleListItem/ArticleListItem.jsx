import React from "react";
import PropTypes from "prop-types";
import ArticleTextToggleButton from "../ArticleTextToggleButton/ArticleTextToggleButton.jsx"
import styles from "./ArticleListItem.module.css"

const ArticleListItem = (props) => {
  return (
    <li className={styles.item}>
      <article>
        <h2 className={styles.title}>{props.article.title}</h2>
        <ArticleTextToggleButton article={props.article}></ArticleTextToggleButton>
      </article>
    </li>
  );
};

ArticleListItem.propTypes = {
  article: PropTypes.object.isRequired,
};
export default ArticleListItem;
