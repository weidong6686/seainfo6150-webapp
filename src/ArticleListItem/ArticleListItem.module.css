
.item {
    box-sizing: border-box;
    box-shadow: 0px 0px 10px #242222 ;
    list-style: none;
    background-image: linear-gradient(to right, #e09292 0%, white 20%);
    border: 1px solid #cc0000;
    border-radius: 10px;
    margin: 20px 15px 15px 15px;
    padding: 0px 10px 10px 10px;
}

.item:first-child {
    margin-top: 0px;
}

.title {
    color: #cc0000;
}