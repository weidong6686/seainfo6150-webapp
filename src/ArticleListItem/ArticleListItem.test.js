import React from "react";
import ArticleListItem from "./ArticleListItem";
import { fireEvent } from "@testing-library/react";

describe("ArticleListItem tests", () => {

  it("renders correctly", () => {
    const article = {
        "title": "Test title", 
        "displayDate": "Test displayDate", 
        "shortText": "Test shortText content"
    };
    const { container } = render(<ArticleListItem article={article}/>);
    expect(container).toMatchSnapshot();
  });
  
});