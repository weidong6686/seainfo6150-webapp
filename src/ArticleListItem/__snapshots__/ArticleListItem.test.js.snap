// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleListItem tests renders correctly 1`] = `
<div>
  <li
    class="container"
  >
    <article
      class="article"
    >
      <div
        class="wrapper"
      >
        <h2
          class="title"
        >
          Test title
        </h2>
      </div>
      <button
        class="button"
      >
        Show more
      </button>
    </article>
  </li>
</div>
`;
